src/tribler-core/tribler_core/session.py

```diff
@@ -456,13 +468,15 @@ async def start(self):
         if self.config.get_libtorrent_enabled():
             self.readable_status = STATE_START_LIBTORRENT
             from tribler_core.modules.libtorrent.download_manager import DownloadManager
-            self.dlmgr = DownloadManager(self)
+            self.dlmgr = DownloadManager(self, dummy_mode=self.core_test_mode)
             self.dlmgr.initialize()
             self.readable_status = STATE_LOAD_CHECKPOINTS
             await self.dlmgr.load_checkpoints()
+            if self.core_test_mode:
+                await self.dlmgr.start_download_from_uri("magnet:?xt=urn:btih:0000000000000000000000000000000000000000")
```

Comment:
I think this start a libtorrent session. Could you check this? If so, can we somehow avoid starting libtorrent during the GUI tests?

Reply:
I'll try to disable starting a real libtorrent session.

Reply:
I managed to `Mock` the libtorrent session in-code